unit_tests/src/test_user_administration.cpp

```diff
@@ -8,6 +8,36 @@
 #include <algorithm>
 #include <iterator>
 
+TEST_CASE("rodsadmin downgrade")
```

Comment:
Writing a C++ unit test is great, but you're going to run into some challenges for this particular case.

Your best option is to switch to the python test suite.

See scripts/irods/test/test_iadmin.py.